### PR TITLE
DOM-api-task

### PR DESCRIPTION
**Pull request by Conversee12**
### DOM-api-task

[Demo](https://conversee12.github.io/DOM-API/) |
[Code base](https://github.com/Conversee12/DOM-API.git)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by Conversee12** [comment]
I've completed self-checks, did'nt find any common errors

File: submissions/Conversee12/DOM/script.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    myBestSong: 'Best song in my opinion : "A Beautiful Lie"',
+    image: "img/mars.jpg"
+  },
+];
+
+function initButtons(container) {
+  return rockBands.forEach((band) => {
+    const { title } = band;
+    const button = document.createElement("button");
+    button.classList.add("buttons");
```
Comment (lysenko-sergey-developer) [inline submissions/Conversee12/DOM/script.js:56]: Why class is `buttons` for a single `button`?

File: submissions/Conversee12/DOM/index.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  <header class="header">
+    <p>Best Hard-Rock Bands 90's</p>
+  </header>
+
+  <main class="main">
+    <ul class="sideBar"></ul>
+    <section class="container">
+      <h2 class="section__title"></h2>
+      <div class="container__image-text">
+        <img src="" alt="" class="section__img">
```
Comment (lysenko-sergey-developer) [inline submissions/Conversee12/DOM/index.html:23]: `img` must have not empty`alt` attribute.

File: submissions/Conversee12/DOM/script.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    button.addEventListener("click", ({ target }) => {
+      showMain(target);
+    });
+    container.append(button);
+  });
+}
+
+function showMain(target) {
+  const content = rockBands.filter(
+    (band) => target.innerText === band.title
```
Comment (lysenko-sergey-developer) [inline submissions/Conversee12/DOM/script.js:67]: Searching by title is a bad idea. Add ids.

File: submissions/Conversee12/DOM/script.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    button.innerText = title;
+    button.addEventListener("click", ({ target }) => {
+      showMain(target);
+    });
+    container.append(button);
+  });
+}
+
+function showMain(target) {
+  const content = rockBands.filter(
```
Comment (lysenko-sergey-developer) [inline submissions/Conversee12/DOM/script.js:66]: Looks like `find` is more right choice in your situation.

File: submissions/Conversee12/DOM/script.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  const bandInfo = document.querySelector(".section__text");
+  const bandSong = document.querySelector(".section__song");
+  bandTitle.innerText = title;
+  bandSong.innerText = myBestSong;
+  bandInfo.innerText = about;
+  bandImage.src = image;
+}
+
+const sideBar = document.querySelector(".sideBar");
+initButtons(sideBar);
```
Comment (lysenko-sergey-developer) [inline submissions/Conversee12/DOM/script.js:81]: You use `sideBar` only inside `initButtons` and `initButtons` call only once. You can move `sideBar` into `initButtons`.

File: submissions/Conversee12/DOM/style.css [hunk 6]
```diff
@@ -0,0 +1,443 @@
...
+  box-sizing: border-box;
+  padding: 0;
+  margin: 0;
+  font-family: "Mochiy Pop One", sans-serif;
+}
+
+body {
+  display: flex;
+  flex-direction: column;
+  background: #0b0e14;
```
Comment (lysenko-sergey-developer) [inline submissions/Conversee12/DOM/style.css:13]: Move colors to css3 vars.

**Comment by lysenko-sergey-developer** [comment]
@Conversee12 Good work! 👍

**Review by lysenko-sergey-developer — CHANGES_REQUESTED** [review]

**Comment by Conversee12** [comment]
Okey, i fixed it, but I didn't understand why (.find) is better here, please explain whether it still needs to be replaced.

**Comment by lysenko-sergey-developer** [comment]
In `showMenu` with `filter`, you extract one rock-group and destruct first element of array to object properties. If you will use `find` you will have one element as result and you don't need to use array destruct.

File: submissions/Conversee12/DOM/script.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+];
+
+function initButtons(container) {
+  return rockBands.forEach((band) => {
+    const { id, title } = band;
+    const button = document.createElement("button");
+    button.classList.add("button");
+    button.id = id;
+    button.innerText = title;
+    button.addEventListener("click", ({ target }) => {
```
Comment (lysenko-sergey-developer) [inline submissions/Conversee12/DOM/script.js:67]: Can we use event bubbling for use single event handler?

**Review by lysenko-sergey-developer — CHANGES_REQUESTED** [review]

File: submissions/Conversee12/DOM/script.js (hunk 7 above)
Comment (Conversee12) [reply submissions/Conversee12/DOM/script.js]: Yee, i finally understand the difference between filter and find, thanks

**Review by lysenko-sergey-developer — APPROVED** [review]
LGTM! 😸

**Pull request merged**